src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,253 @@
+package duke.parser;
+
+import duke.exception.IncompleteCommandException;
+import duke.exception.IndexOffBoundException;
+import duke.exception.InvalidCommandException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * parser class that deals with making sense of the user command.
+ */
+public class Parser {
+    private TaskList taskList;
+    private Scanner sc;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Initialise parser class with TaskList and ui class.
+     *
+     * @param taskList list of tasks
+     * @param ui       object that deals with interaction with user
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        sc = new Scanner(System.in);
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Check if the user command input is valid,
+     * otherwise it will prompt user to give a
+     * valid input again.
+     *
+     * @return a list of the updated tasks
+     */
+    public String parse(String input) {
```

Comment:
Perhaps for this method, you could break down the code within each if statements and make methods to handle each chunk of code? This would improve the readability of the code and make it more OOP overall, in my opinion.